auto-update.php

```diff
@@ -0,0 +1,113 @@
+	}
+
+	/**
+	 * Adds the wc-api-dev plugin to the list of allowed plugins for auto update.
+	 */
+	function auto_update( $update, $item ) {
+		if ( in_array( $item->slug, array( 'wc-api-dev' ) ) ) {
+			return true;
+		} else {
+			return $update;
+		}
+	}
+
+	/**
+	 * Fetch the first non pre-release zip from GitHub releases and store the response for later use.
+	 */
+	private function maybe_fetch_github_response() {
+		if ( is_null( $this->github_response ) ) {
+			$request_uri = 'https://api.github.com/repos/woocommerce/wc-api-dev/releases';
+			$response = json_decode( wp_remote_retrieve_body( wp_remote_get( $request_uri ) ), true );
+			if ( is_array( $response ) ) {
+				foreach( $response as $entry ) {
+					if ( false === ( bool ) $entry['prerelease'] ) {
+						$this->github_response = $entry;
+						break;
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Add our plugin to the list of plugins to update, if we find the version is out of date.
+	 */
+	public function modify_transient( $transient ) {
+		if( property_exists( $transient, 'checked' ) && $transient->checked ) {
+			$checked = $transient->checked;
+			$this->maybe_fetch_github_response();
+			$out_of_date = version_compare( $this->github_response['tag_name'], $checked[ $this->file ], 'gt' );
+			if( $out_of_date ) {
+				$plugin = array(
+					'url'         => 'https://github.com/woocommerce/wc-api-dev',
+					'plugin'      => $this->file,
+					'slug'        => 'wc-api-dev',
+					'package'     => $this->github_response['zipball_url'],
+					'new_version' => $this->github_response['tag_name']
+				);
+				$transient->response[ $this->file ] = (object) $plugin;
+			}
+		}
+		return $transient;
+	}
+
+	/**
+	 * Displays some basic information from our README if the user tries to
+	 * view plugin info for the update via the manual update screen.
+	 */
+	public function plugin_popup( $result, $action, $args ) {
+		if ( ! empty( $args->slug ) ) {
+			if ( $args->slug === current( explode( '/' , $this->file ) ) ) {
+				$this->maybe_fetch_github_response();
+				$plugin = get_plugin_data( dirname( dirname( __FILE__ ) ) . '/' . $this->file );
+				$plugin = array(
+					'name'              => $plugin['Name'],
+					'slug'              => $this->file,
+					'version'           => $this->github_response['tag_name'],
+					'author'            => $plugin['AuthorName'],
+					'author_profile'    => $plugin['AuthorURI'],
+					'last_updated'      => $this->github_response['published_at'],
+					'homepage'          => $plugin['PluginURI'],
+					'short_description' => $plugin['Description'],
+					'sections'          => array(
+						'Description'   => $plugin['Description'],
+						'Updates'       => $this->github_response['body'],
+					),
+					'download_link'     => $this->github_response['zipball_url']
+				);
+				return (object) $plugin;
+			}
+		}
+		return $result;
+	}
+
+	/**
+	 * Move the updated plugin to the correct directory, and reactivate the plugin.
```

Comment:
Please include a comment as to why this is necessary. I assume because the Github zip has a short hash in the filename?